### PR TITLE
Backport modal listener key change

### PR DESCRIPTION
**Pull request by taion**
cc @jeffredodd

### CODE REVIEW

**Comment by taion** [comment]
@jquense Quick review on this? I didn't realize React-Bootstrap wasn't using v0.8.x, and this will allow people to actually see the modal a11y fix, without making us release `next`.

File: src/Modal.js [hunk 1]
```diff
@@ -440,12 +456,21 @@ class Modal extends React.Component {
...
+        this.props.onEscapeKeyDown(e);
+      }
+
+      this.props.onHide();
+    }
+  }
+
   handleDocumentKeyUp = (e) => {
-    if (this.props.keyboard && e.keyCode === 27 && this.isTopModal()) {
+    if (this.props.keyboard && e.key === 'Escape' && this.isTopModal()) {
```
Comment (jeffredodd) [inline src/Modal.js]: This is a small question I could probably google and get the answer but maybe someone already knows. Will this key always be `'escape'` even on a Chinese keyboard for example? Just concerned about localization when hard coding english words in the codebase.

Comment (taion) [reply src/Modal.js]: Yup, see https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key/Key_Values

(Note `Escape`, never `escape` 😛)

**Comment by taion** [comment]
Released in v0.7.1.

**Comment by taion** [comment]
(so just reinstall deps for React-Bootstrap as needed and you should be good)

**Pull request merged**